src/test/java/org/sagebionetworks/web/unitclient/widget/entity/controller/EntityActionControllerImplTest.java

```diff
@@ -2600,29 +2854,38 @@ public void testOnDeleteConfirmedPreFlightPassedDeleteFailed() {
     // the call under test
     controller.onAction(Action.DELETE_ENTITY, null);
     verify(mockView).showConfirmDeleteDialog(anyString(), any(Callback.class));
-    verify(mockPreflightController)
-      .checkDeleteEntity(any(EntityBundle.class), any(Callback.class));
+    verify(mockPreflightController).checkDeleteEntity(
+      any(EntityBundle.class),
+      any(Callback.class)
+    );
     // an attempt to delete should be made
-    verify(mockSynapseJavascriptClient)
-      .deleteEntityById(anyString(), any(AsyncCallback.class));
-    verify(mockView)
-      .showErrorMessage(DisplayConstants.ERROR_ENTITY_DELETE_FAILURE + error);
+    verify(mockSynapseJavascriptClient).deleteEntityById(
+      anyString(),
+      any(AsyncCallback.class)
+    );
+    verify(mockView).showErrorMessage(
+      DisplayConstants.ERROR_ENTITY_DELETE_FAILURE + error
+    );
+    QueryKey mockQueryKey = mock(QueryKey.class);
+    when(mockKeyFactory.getTrashCanItemsQueryKey()).thenReturn(mockQueryKey);
+    verify(mockKeyFactoryProvider, never()).getKeyFactory(anyString());
+    verify(mockKeyFactory, never()).getTrashCanItemsQueryKey();
+    verify(mockQueryClient, never()).invalidateQueries(
+      any(InvalidateQueryFilters.class)
+    );
```

Comment:
Test to check that trashcan cache is not invalidated if the deletion is unsuccessful